lib/gstreamer/va/vpp.py

```diff
@@ -56,6 +56,8 @@ def gen_vpp_opts(self):
       opts += " {vpp_op}={mlevel}"
     elif self.vpp_op in ["transpose"]:
       opts += " video-direction={direction}"
+    elif self.vpp_op in ["crop"]:
+      opts += " disable-passthrough=true ! videocrop left={left} right={right} top={top} bottom={bottom}"
```

Comment:
The `vapostproc` does not handle cropping for the **downstream** `vidoecrop` element.  Here, the downstream `videocrop` is doing the actual cropping which is not what we should be testing.   The `vapostproc` only handles cropping from an **upstream** `videocrop` element.  

To get the `videocrop` before the `vapostproc`, it will need to be done in the `gen_output_opts` override just before the _super()_ implementation is called.

There is a bug reported at https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/2653.  Thus, there will be failing tests... but that is ok.  It will put pressure to get it fixed in gstreamer.

Reply:
updated: 
With new result shown as below:
Session ended. 8 successful, 0 skipped, 8 failed, 0 erroneous. Total duration: 00:00:04 

Currently vaapi-fits cmdline changed to:
gst-launch-1.0 -vf filesrc location=/opt/assets/otc-media/sanity/yuv/1080p_blue_sky_1920x1080_NV12.yuv num-buffers=10 ! rawvideoparse format=nv12 width=1920 height=1080 !  videocrop left=10 right=0 top=0 bottom=0 ! vapostproc  disable-passthrough=true ! video/x-raw,format=NV12 ! checksumsink2 file-checksum=false qos=false frame-checksum=false plane-checksum=false dump-output=true dump-location=./939960a8-0b1f-11ee-95c0-0242ac120002_0/_0.test.gst-va.vpp.crop/default/939960a8-0b1f-11ee-95c0-0242ac120002_0_1_1.yuv eos-after=10